src/components/trade/PriceSuggestions/PriceSuggestionItem.tsx

```diff
@@ -26,20 +27,34 @@ interface FormattedPrices {
   inversePriceValue: string
 }
 
-const LOW_PRICE_FLOOR = new BigNumber('0.0001')
-
-export function formatPriceToPrecision(price: BigNumber, useThreshold = false): string {
-  return price.gt(LOW_PRICE_FLOOR) || !useThreshold
-    ? price.toFixed(PRICE_ESTIMATION_PRECISION)
-    : '< ' + LOW_PRICE_FLOOR.toString()
+interface GetPriceFormatted {
+  price: BigNumber | null
+  isPriceInverted: boolean
+  baseTokenDecimals: number
+  quoteTokenDecimals: number
 }
 
-function getPriceFormatted(price: BigNumber | null, isPriceInverted: boolean): FormattedPrices {
+function getPriceFormatted({
+  price,
+  isPriceInverted,
+  baseTokenDecimals,
+  quoteTokenDecimals,
+}: GetPriceFormatted): FormattedPrices {
   if (price) {
-    const inversePriceLabel = formatPriceToPrecision(invertPrice(price), true)
-    const priceLabel = formatPriceToPrecision(price, true)
-    const inversePriceValue = invertPrice(price).toString(10)
-    const priceValue = price.toString(10)
+    const inversePriceLabel = formatPriceWithFloor(invertPrice(price))
+    const priceLabel = formatPriceWithFloor(price)
+
+    // Use long precision form for accuracy
+    // Use BigNumber's decimalPlaces(<# of decimal places to show>, ROUNDING_MODE)
+    // Rounds away 0's
+
+    // ROUNDING_MODE [1] => Rounds towards zero
+    // 1. new BigNumber(0.0016600425).decimalPlaces(9,1).toString(10) => "0.001660042"
+    // 2. new BigNumber(0.0016600000).decimalPlaces(9,1).toString(10) => "0.00166"
```

Comment:
If u add an example, good to add simpler example. 9 decimals is too much :)

Reply:
OK will add easier example, and for second point, yeah exactly

Reply:
toFixed would leave padding